### ISSUE TITLE
Getting error while fetching most of the tables

### ISSUE THREAD

**Issue by raybittuiq** [turbot/steampipe-plugin-snowflake#77]
**Describe the bug**
Getting error while querying tables, I attaching all the tables and the respective error I am getting while querying, I am using Postgres 15 FDW

**Steampipe version (`steampipe -v`)**
Using postgres15 FDW

**Plugin version (`steampipe plugin list`)**
Example: v0.8.0

snowflake_schemata   -   
	ERROR:  snowflake: rpc error: code = Internal desc = hydrate function listSnowflakeSchemata failed with panic unknown column OWNER_ROLE_TYPE

snowflake_view_grant  -   
 	ERROR:  snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_view_grant' is missing 3 required quals:
    column:'view_name' operator: =
    column:'database_name' operator: =
    column:'schema_name' operator: =

snowflake_session    -
	ERROR:  snowflake: rpc error: code = Internal desc = hydrate function listSnowflakeSession failed with panic unknown column CLOSED_REASON

snowflake_user       -
	ERROR:  snowflake: sql: expected 28 destination arguments in Scan, not 26
	
snowflake_view     -
	ERROR:  snowflake: sql: expected 12 destination arguments in Scan, not 10

snowflake_database_grant     -
	ERROR:  snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_database_grant' is missing 1 required qual: column:'database' operator: =

snowflake_user_grant		-
	ERROR:  snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_user_grant' is missing 1 required qual: column:'username' operator: =

snowflake_role_grant		-
	ERROR:  snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_role_grant' is missing 1 required qual: column:'role' operator: =

snowflake_warehouse         -
	ERROR:  snowflake: missing destination name budget in *[]snowflake.Warehouse

snowflake_account_grant		-
	ERROR:  snowflake: sql: expected 9 destination arguments in Scan, not 8

snowflake_database          -
	ERROR:  snowflake: sql: expected 12 destination arguments in Scan, not 9

snowflake_login_history		-
	ERROR:  snowflake: rpc error: code = Internal desc = hydrate function listSnowflakeLoginHistory failed with panic unknown column CONNECTION

snowflake_network_policy	-
	ERROR:  snowflake: sql: expected 7 destination arguments in Scan, not 5

**Comment by ParthaI** [comment]
Hello, @raybittuiq. I apologize for any inconvenience this may have caused.

The error message you’re seeing, which follows the pattern `<table_name> is missing <number of columns> required qual...`, is expected. This occurs because, based on the API behavior, the tables require specific query parameters to be included in the WHERE clause to successfully make the underlying API call. It seems we may have overlooked including this information in the table documentation. I will update the documentation to clarify which tables require these query parameters to populate the column values correctly.

As for the other errors you’ve encountered, I’ll review them and provide you with an update soon.

Thank you!

**Comment by raybittuiq** [comment]
@ParthaI I tried running the query mentioned in the readme

```
select
  name,
  email,
  disabled,
  last_success_login
from
  snowflake_user
where
  (last_success_login > now() - interval '30 days')
  and last_success_login is not null;
```

but it also throwed that error

**Comment by ParthaI** [comment]
@raybittuiq, In my previous comment, I mentioned that the error like `ERROR: snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_database_grant' is missing 1 required qual: column:'database' operator: =` is expected to get. 

If you want to get rid of this type of error please include the query parameter in where clause like `select * from snowflake_database_grant where database = 'abc'`

I will also review the following errors:

- `pq: public: sql: expected 28 destination arguments in Scan, not 26`
- `ERROR: snowflake: missing destination name 'budget' in *[]snowflake.Warehouse`
- `ERROR: snowflake: rpc error: code = Internal desc = hydrate function listSnowflakeSchemata failed with panic unknown column OWNER_ROLE_TYPE`

I'll investigate these issues further and provide an update.

**Comment by raybittuiq** [comment]
@ParthaI Yes, I tried that as well, and the error like` ERROR: snowflake: rpc error: code = Internal desc = 'List' call for table 'snowflake_database_grant' is missing 1 required qual: column:'database' operator: =` went away but in the same query I got error like `pq: public: sql: expected 28 destination arguments in Scan, not 26`

**Comment by ParthaI** [comment]
Okay, thanks for the confirmation I will take a look.

**Comment by ParthaI** [comment]
Hi @raybittuiq, FYI, I have raised [a PR](https://github.com/turbot/steampipe-plugin-snowflake/pull/78) to address the above-mentioned issue. I hope the changes made in the PR will resolve the issues.

Thanks!

**Comment by dongho-jung** [comment]
It's been over a month since I've been unable to run any queries related to Snowflake, and I was hoping to get an update on the progress.

May I kindly ask if we're just waiting for approval, or if there are still any remaining areas that need to be reviewed?

**Comment by ParthaI** [comment]
Apologies for the delay in the review process, @dongho-jung. @misraved, could you please take a look at this PR?

**Comment by misraved** [comment]
Apologies @dongho-jung , I will be taking up this PR on priority and will try to get a new plugin version released by tomorrow EOD.

Thanks!!

**Comment by dongho-jung** [comment]
I'm sorry, but is there a way to install it, even just the state of the main branch, before it's officially released?

I had an asset management logic that integrated Snowflake with Steampipe, but it has been broken for a long time. I'm wondering if there is a way for me to apply the updates myself before the new version is officially released.

Thank you always for your contributions to open source. If there's any part where I can assist, please let me know.

**Comment by misraved** [comment]
Hi @dongho-jung, the fixes are now available in `v0.9.0` plugin version 👍 !!

I'm sorry for the delay.
